packages/loader/container-loader/src/container.ts

```diff
@@ -517,6 +517,15 @@ export class Container
 				0x969 /* not connected yet */,
 			);
 
+			// Track membership changes and update connection state accordingly
+			// We do this call here, instead of doing it in initializeProtocolState() due to pendingLocalState.
+			// When we load from stashed state, we let connectionStateHandler know about clientId from previous container instance.
+			// But we will play trailing ops from stanshot, including potentially playing join & leave ops for that same clientId!
+			// IN other words, if connectionStateHandler has access to Quorum early in load sequence, it will see events (in stashed ops mode)
+			// in the order that is not possible in real life, that it may not expect.
+			// Ideally, we should supply pendingLocalState?.clientId here as well, not in constructor, but it does not matter (at least today)
+			this.connectionStateHandler.initProtocol(this.protocolHandler);
```

Comment:
This makes sense, but wanted to think out loud about it too since this kind of move is pretty subtle.

I looked at all the code that runs between the old spot and new spot.  The critical piece is obviously replaying stashed ops ("saved ops"), but it also will do ops fetch. Any concerns with delaying this past that point?

What I can think of is that _other_ clients' join/leave ops could come in, but those never matter to ConnectionStateHandler, it is only concerned with its own.

And this function is all about readying the container to be connected, so it makes sense to go here.

Reply:
Correct. There is some risk involved here, and to be honest - I'd rather look (over time) at keeping this call where it was, but change when / how we communicate pendingLocalState.clientId to connectionStateHandler (make it know about such clientId much later in load sequence). I'll look more into it, maybe it's not that hard - not sure.